src/node/mini_miner.cpp

```diff
@@ -201,8 +245,9 @@ void MiniMiner::SanityCheck() const
         [&](const auto& txid){return m_entries_by_txid.find(txid) == m_entries_by_txid.end();}));
 }
 
-void MiniMiner::BuildMockTemplate(const CFeeRate& target_feerate)
+void MiniMiner::BuildMockTemplate(std::optional<CFeeRate> target_feerate)
 {
+    uint32_t sequence_num{0};
     while (!m_entries_by_txid.empty()) {
         // Sort again, since transaction removal may change some m_entries' ancestor feerates.
```

Comment:
When a transaction's parent is mined and it isn't, its parent is no longer part of its ancestor fees and size. This may change its ancestor feerate.

The analogous concept in `BlockAssembler` is [`mapModifiedTx`](https://github.com/bitcoin/bitcoin/blob/64879f4c03b9d933898c0bb83d9b7e9c20729e51/src/node/miner.cpp#L297-L299).

Btw, there are [review club notes](https://bitcoincore.reviews/26152) that may be helpful for understanding #27021.